model/riscv_insts_dext.sail

```diff
@@ -251,25 +254,25 @@ function validDoubleRegs(n, regs) = {
 
 /* AST */
 
-union clause ast = F_MADD_TYPE_D : (regidx, regidx, regidx, rounding_mode, regidx, f_madd_op_D)
+union clause ast = F_MADD_TYPE_D : (fregidx, fregidx, fregidx, rounding_mode, fregidx, f_madd_op_D)
 
 /* AST <-> Binary encoding ================================ */
 
 mapping clause encdec =
     F_MADD_TYPE_D(rs3, rs2, rs1, rm, rd, FMADD_D)                         if haveDoubleFPU() & validDoubleRegs([rs3, rs2, rs1, rd])
-<-> rs3 @ 0b01 @ rs2 @ rs1 @ encdec_rounding_mode (rm) @ rd @ 0b100_0011  if haveDoubleFPU() & validDoubleRegs([rs3, rs2, rs1, rd])
+<-> encdec_freg(rs3) @ 0b01 @ encdec_freg(rs2) @ encdec_freg(rs1) @ encdec_rounding_mode (rm) @ encdec_freg(rd) @ 0b100_0011 if haveDoubleFPU() & validDoubleRegs([rs3, rs2, rs1, rd])
```

Comment:
It would be good to preserve the  alignment of the `if` conditionals on either side of `<->` to help visually ensure the conditions are the same, even though it does add an extra line to the diff.

Reply:
That was only inconsistently the case beforehand, but I think I have improved it.